buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -203,21 +206,35 @@ private static int findDefaultParallel(Project project) {
         return _defaultParallel;
     }
 
-    private String gitRevision(final Project project) {
-        final ByteArrayOutputStream stdout = new ByteArrayOutputStream();
-        final ByteArrayOutputStream stderr = new ByteArrayOutputStream();
-        final ExecResult result = project.exec(spec -> {
-            spec.setExecutable("git");
-            spec.setArgs(Arrays.asList("rev-parse", "HEAD"));
-            spec.setStandardOutput(stdout);
-            spec.setErrorOutput(stderr);
-            spec.setIgnoreExitValue(true);
-        });
+    private Object createGitRevisionResolver(final Project project) {
+        return new Object() {
+            private final AtomicReference<String> gitRevision = new AtomicReference<>();
```

Comment:
Would it be better instead of making this an `AtomicReference` to simply synchronize these calls so we can avoid to possibility shelling to `git` multiple times if this is called concurrently? I think as it is we could do that and simply the first one would win.

Reply:
While this is a possibility, I wasn't sure it was necessary to avoid. I figured it was better to make concurrent calls fast, if they happen (but it seems unlikely?). By synchronizing the method, we would still be single threading the method even after we've retrieved the hash.

Reply:
Yeah, it's hard to say either way. I'm fine with leaving as is.